native/core/src/parquet/parquet_exec.rs

```diff
@@ -0,0 +1,141 @@
+///
+///   `required_schema`: Schema to be projected by the scan.
+///
+///   `data_schema`: Schema of the underlying data. It is optional and, if provided, is used
+/// instead of `required_schema` to initialize the file scan
+///
+///   `partition_schema` and `partition_fields` are optional. If `partition_schema` is specified,
+/// then `partition_fields` must also be specified
+///
+///   `object_store_url`: Url to read data from
+///
+///   `file_groups`: A collection of groups of `PartitionedFiles` that are to be read by the scan
+///
+///   `projection_vector`: A vector of the indexes in the schema of the fields to be projected
+///
+///   `data_filters`: Any predicate that must be applied to the data returned by the scan. If
+/// specified, then `data_schema` must also be specified.
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn init_parquet_exec(
+    required_schema: SchemaRef,
+    data_schema: Option<SchemaRef>,
+    partition_schema: Option<SchemaRef>,
+    partition_fields: Option<Vec<Field>>,
+    object_store_url: ObjectStoreUrl,
+    file_groups: Vec<Vec<PartitionedFile>>,
+    projection_vector: Option<Vec<usize>>,
+    data_filters: Option<Vec<Arc<dyn PhysicalExpr>>>,
+    session_timezone: &str,
+) -> Result<Arc<DataSourceExec>, ExecutionError> {
+    let (table_parquet_options, spark_parquet_options) = get_options(session_timezone);
+    let mut parquet_source = ParquetSource::new(table_parquet_options).with_schema_adapter_factory(
+        Arc::new(SparkSchemaAdapterFactory::new(spark_parquet_options)),
+    );
+    // Create a conjunctive form of the vector because ParquetExecBuilder takes
+    // a single expression
+    if let Some(data_filters) = data_filters {
+        let cnf_data_filters = data_filters.clone().into_iter().reduce(|left, right| {
+            Arc::new(BinaryExpr::new(
+                left,
+                datafusion::logical_expr::Operator::And,
+                right,
+            ))
+        });
+
+        if let (Some(filter), Some(data_schema)) = (cnf_data_filters, &data_schema) {
+            parquet_source = parquet_source.with_predicate(Arc::clone(data_schema), filter);
+        }
+    }
+    let file_scan_config = match (data_schema, projection_vector, partition_fields) {
+        (Some(data_schema), Some(projection_vector), Some(partition_fields)) => get_file_config(
+            data_schema,
+            partition_schema,
+            file_groups,
+            object_store_url,
+            Arc::new(parquet_source),
+        )
+        .with_projection(Some(projection_vector))
+        .with_table_partition_cols(partition_fields),
+        _ => get_file_config(
+            required_schema,
+            partition_schema,
+            file_groups,
+            object_store_url,
+            Arc::new(parquet_source),
+        ),
+    };
+
+    let scan = Arc::new(DataSourceExec::new(Arc::new(file_scan_config)));
+
+    Ok(scan)
```

Comment:
The variable `scan` isn't really required here.

```suggestion
    Ok(Arc::new(DataSourceExec::new(Arc::new(file_scan_config))))
```